components/layout_2020/flow/mod.rs

```diff
@@ -360,59 +360,78 @@ fn calculate_inline_content_size_for_block_level_boxes(
     boxes: &[ArcRefCell<BlockLevelBox>],
     layout_context: &LayoutContext,
     containing_block: &IndefiniteContainingBlock,
-) -> ContentSizes {
+) -> InlineContentSizesResult {
     let get_box_info = |box_: &ArcRefCell<BlockLevelBox>| {
         match &mut *box_.borrow_mut() {
             BlockLevelBox::OutOfFlowAbsolutelyPositionedBox(_) |
             BlockLevelBox::OutsideMarker { .. } => None,
             BlockLevelBox::OutOfFlowFloatBox(ref mut float_box) => {
-                let size = float_box
-                    .contents
-                    .outer_inline_content_sizes(
-                        layout_context,
-                        containing_block,
-                        &LogicalVec2::zero(),
-                        false, /* auto_block_size_stretches_to_containing_block */
-                    )
-                    .max(ContentSizes::zero());
+                let InlineContentSizesResult {
+                    depends_on_block_constraints,
+                    sizes,
+                } = float_box.contents.outer_inline_content_sizes(
+                    layout_context,
+                    containing_block,
+                    &LogicalVec2::zero(),
+                    false, /* auto_block_size_stretches_to_containing_block */
+                );
```

Comment:
Maybe it'd be a little bit less code to return `InlineContentSizesResult` in the tuple here. It could be made mutable and the sizes could be adjusted with `max_assign(...)`. It looks like currently `sizes.max(ContentSizes::zero()),` is executed for every branch of the match that produces a value?

Reply:
Done